ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -563,6 +563,29 @@ class ClientQueryViewController: UITableViewController, Themeable {
 
 		self.present(createFolderNavigationVC, animated: true, completion: viewDidAppearHandler)
 	}
+
+	func duplicate(_ item: OCItem, viewDidAppearHandler: ClientActionVieDidAppearHandler? = nil, completionHandler: ClientActionCompletionHandler? = nil) {
+		var name: String = "\(item.name!) copy"
+
+		if item.type != .collection {
+			let suffix = item.fileSuffix()
+			let itemNameWithouExtension = item.name.dropLast(suffix.count + 1) // The extra one is for the '.' in the extension.
```

Comment:
I would rather have a method in OCItem returning the name w/o extension instead of in-place string manipulation for better readability. Also what is with files without extension? Those are pretty common in the Apple ecosystem.

Reply:
I'll create a new function in `OCItem+Extension` like:
```Swift
func nameWithoutExtension() -> String {
	return (self.name as NSString).deletingPathExtension
}
```

>Also what is with files without extension? 
by using `NSString` APIs we can use `pathExtension` [NSString pathExtension](https://developer.apple.com/documentation/foundation/nsstring/1407801-pathextension)

if there is no extension it return `""` so should not be any problem with files without extension.

Reply:
@pablocarmu still you would assume that there is a "." character in dropLast() call, right?

Reply:
I'll change the implementation of this part to:
```Swift
if item.type != .collection {
	let itemName = item.nameWithoutExtension()
	var fileExtension = item.fileExtension()

	if fileExtension != "" {
		fileExtension = ".\(fileExtension)"
	}

	name = "\(itemName) copy\(fileExtension)"
}
```
I can do it here or I can change `fileExtension()` implementation to return `""` in case there is no extension or `.extension` if there is. I think the second option is the way to go, do you agree?